jenkinsapi/build.py

```diff
@@ -53,7 +53,18 @@ def __init__(self, url, buildno, job, depth=1):
         self.buildno = buildno
         self.job = job
         self.depth = depth
-        JenkinsBase.__init__(self, url)
+
+        if job.get_jenkins_obj().base_server_url() and job.get_jenkins_obj().get_use_baseurl():
+            build_hostname = urlparse(url).netloc
+            jenkins_obj_hostname = urlparse(job.get_jenkins_obj().base_server_url()).netloc
+            if build_hostname != jenkins_obj_hostname:
+                real_url = url.replace(build_hostname, jenkins_obj_hostname)
+            else:
```

Comment:
Is this `else` statement necessary here? It seems like the one on line 64 would catch and assign `real_url = url`.

Reply:
The else would be executed if the hostname in job url and jenkins url are the same. Adding test cases to test that path